eqcorrscan/utils/plotting.py

```diff
@@ -2186,6 +2186,131 @@ def subspace_fc_plot(detector, stachans, **kwargs):
     return fig
 
 
+@additional_docstring(plotting_kwargs=plotting_kwargs)
+def mapplot(events, bgcolor='#909090', method='depth', cpalette='jet_r',
```

Comment:
Specifying these defaults here will be problematic if the user passes additional plotting arguments as a dictionary.  I would recommend setting the defaults in the code and using them if things like `s`, `lw`, `marker`, etc are not given by the user as `kwargs`.   That way you can pass `kwargs` to the underlying matplotlib functions and we can support passing through all the standard matplotlib customization arguments.

Reply:
Ok.

Reply:
Hi.
I wrote these defaults as kwargs and it made a problem.
When we define any parameter like savefile or something else as input of the function, it raise an error in `scatter` function.
Then we can't pass any kwargs to `_finalise_figure`.
I changed name of the function to twoD_seismplot.
```python
Traceback (most recent call last):
  File "plotting.py", line 234, in <module>
    twoD_seismplot(locations=locations, savefile='1.png', show=False)
  File "plotting.py", line 168, in twoD_seismplot
    map0 = ax0.scatter(lon, lat, c=c0, **kwargs)
  File "/home/imon/.local/lib/python3.6/site-packages/matplotlib/__init__.py", line 1565, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/home/imon/.local/lib/python3.6/site-packages/matplotlib/cbook/deprecation.py", line 358, in wrapper
    return func(*args, **kwargs)
  File "/home/imon/.local/lib/python3.6/site-packages/matplotlib/axes/_axes.py", line 4432, in scatter
    collection.update(kwargs)
  File "/home/imon/.local/lib/python3.6/site-packages/matplotlib/artist.py", line 1006, in update
    ret = [_update_property(self, k, v) for k, v in props.items()]
  File "/home/imon/.local/lib/python3.6/site-packages/matplotlib/artist.py", line 1006, in <listcomp>
    ret = [_update_property(self, k, v) for k, v in props.items()]
  File "/home/imon/.local/lib/python3.6/site-packages/matplotlib/artist.py", line 1002, in _update_property
    .format(type(self).__name__, k))
AttributeError: 'PathCollection' object has no property 'savefile'
```

Reply:
Ah, fair enough - you would have to make a new dictionary of kwargs for `_finalise_figure` within the function - sorry, just leave that to me in a later redesign and I will try to apply it homogeneously to everything.

Thanks for the rename - I'm not fond of the capital D, but that is my fault as well - maybe down the line we can rename these (this and `threeD...`).